File: valid-parentheses/valid-parentheses.js
/**
 * @param {string} s
 * @return {boolean}
 */
var isValid = function(s) {
  const map = {
      ')': '(',
      ']': '[',
      '}': '{'        
  };
  const open = ['(', '{', '['];
  const stack = [];
  for(let c of s) {
      if (open.includes(c)) {
          stack.push(c);
          continue;
      }
      const t = stack.pop();
      if (map[c] !== t) {
          return false;
      }
      
  }    
  return stack.length == 0;    
};

File: valid-parentheses/valid-parentheses.go
func isValid(s string) bool {
    
    p := map[rune]rune{
		'(': ')',
		'{': '}',
		'[': ']',
    }
    
    stack := []rune{};

    for _, ch := range s {
        next, ok := p[ch];
        if ok {
            stack = append(stack, next)
            continue;
        }
        n := len(stack) - 1;
        if n < 0 {
            return false
        }
        t := stack[n]
        stack = stack[:n]
        if ch != t {
            return false
        }
    }
    return len(stack) == 0
}